File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/job/meta/KillJobResponse.scala
package com.webank.eggroll.core.deepspeed.job.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class KillJobResponse(sessionId: String)

object KillJobResponse {
  implicit def serialize(src: KillJobResponse): Array[Byte] = {
    Deepspeed.KillJobResponse.newBuilder()
      .setSessionId(src.sessionId)
      .build()
      .toByteArray
  }

  implicit def deserialize(byteString: ByteString): KillJobResponse = {
    val proto = Deepspeed.KillJobResponse.parseFrom(byteString)
    KillJobResponse(proto.getSessionId)
  }
}

File: jvm/core/src/main/java/com/eggroll/core/pojo/KillJobResponse.java
package com.eggroll.core.pojo;

import com.webank.eggroll.core.meta.Deepspeed;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class KillJobResponse implements RpcMessage {

    Logger log = LoggerFactory.getLogger(KillJobResponse.class);

    private String sessionId;

    @Override
    public byte[] serialize() {
        Deepspeed.KillJobResponse.Builder builder = Deepspeed.KillJobResponse.newBuilder();
        builder.setSessionId(this.sessionId);
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {
        try {
            Deepspeed.KillJobResponse killJobResponse = Deepspeed.KillJobResponse.parseFrom(data);
            this.sessionId = killJobResponse.getSessionId();
        } catch (Exception e) {
            log.error("deserialize error : ", e);
        }
    }
}
